app/renderer/components/frame/frame.js

```diff
@@ -682,16 +682,17 @@ class Frame extends React.Component {
       if (e.securityState === 'secure') {
         isSecure = true
       } else if (e.securityState === 'insecure') {
-        isSecure = false
+        // Passive mixed content should not upgrade an insecure connection to a
+        // partially-secure connection. It can only downgrade a secure
+        // connection.
+        isSecure =
+          e.securityInfo.mixedContentStatus === 'content-status-displayed' && this.props.isSecure !== false
```

Comment:
do we need to handle `content-status-displayed-and-ran` which indicates both passive and active mixed content?

Reply:
never mind, I forgot we default block that active mixed content scripts so it won't reach that case

Reply:
yeah, i tried allowing insecure scripts on https://very.badssl.com/, but it only emitted `content-status-displayed`